pyvista/core/dataset.py

```diff
@@ -1187,6 +1192,9 @@ def scale(
         >>> _ = pl.add_mesh(mesh2)
         >>> pl.show(cpos="xy")
         """
+        if isinstance(xyz, (float, int, np.number)):
+            xyz = [xyz] * 3
+
```

Comment:
Tests were failing because the scalar case used to work due to array broadcasting from scalars to arbitrary shape. I've added this back, and extended the type hints a bit, please check.